SingularityUI/app/components/common/modalButtons/RedeployButton.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { Glyphicon } from 'react-bootstrap';
+import OverlayTrigger from 'react-bootstrap/lib/OverlayTrigger';
+import ToolTip from 'react-bootstrap/lib/Tooltip';
+
+import { getClickComponent } from '../modal/ModalWrapper';
+import { FetchDeployForRequest } from '../../../actions/api/history';
+
+import RedeployModal from './RedeployModal';
+import Utils from '../../../utils';
+
+const redeployTooltip = (
+  <ToolTip id="redeploy">
+    Redeploy
+  </ToolTip>
+);
+
+class RedeployButton extends Component {
```

Comment:
This is how I did it: https://github.com/HubSpot/Singularity/blob/0e65a7407b9c1a9eee0417d20aabd8d68ceb5b2d/SingularityUI/app/components/common/modalButtons/NewWebhookButton.jsx

That doesn't help with your dropdown problem though. (Anyway, remember @markhazlewood doesn't like dropdowns - I agree with his reasoning as to why)